app/rest/jsondb/src/test/java/io/syndesis/jsondb/impl/JsonDBTest.java

```diff
@@ -587,6 +594,108 @@ public void shouldAscertainPropertyPairExistence() {
         assertThat(jsondb.fetchIdsByPropertyValue("/pair", "key", "nope")).isEmpty();
     }
 
+
+    @Test
+    public void testGetFilter() throws IOException {
+
+        jsondb.set("/users/u1", mapper.writeValueAsString(map(
+            "name", "u1",
+            "age", 9
+        )));
+        jsondb.set("/users/u2", mapper.writeValueAsString(map(
+            "name", "u2",
+            "age", 10
+        )));
+        jsondb.set("/users/u3", mapper.writeValueAsString(map(
+            "name", "u3",
+            "age", 21
+        )));
+
+        ////////////////////////////////////////
+        //
+        // Check filtering ops against strings..
+        //
+        ////////////////////////////////////////
+        assertThat(jsondb.getAsString("/users", new GetOptions().filter(
```

Comment:
might be a good idea to use [JSONAssert](http://jsonassert.skyscreamer.org/)

Reply:
I would reduce test coverage.  We want to verify the order of things that jsondb returns.